src/bridge/cockpitpackages.c

```diff
@@ -630,7 +651,15 @@ maybe_add_package (GHashTable *listing,
   directory = NULL;
 
   if (own_checksum)
-    package->own_checksum = g_strdup (g_checksum_get_string (own_checksum));
+    {
+       /* digest the whole final manifest, which may have overrides from external directories */
+      gsize manifest_len;
+      g_autofree gchar *manifest_str = cockpit_json_write_object (manifest, &manifest_len);
+      g_checksum_update (own_checksum, (guchar *) manifest_str, manifest_len);
```

Comment:
I'd still prefer to see this done a little bit more like the other files: possibly add `"manifest.json\0"` before, and definitely include the nul byte.

Reply:
Does that make any practical difference? Adding constant strings into a checksum just feels like a waste of time. No strong opinion though, I can update this once we unbreak journal tests on main.

Reply:
Ya.  Fair enough.  this is always at the end, and we always do it exactly once.